internal/manifestvalues/validate_test.go

```diff
@@ -0,0 +1,78 @@
+package manifestvalues
+
+import (
+	"github.com/glasskube/glasskube/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ValidateResolvedValues", func() {
+	emptyManifest := v1alpha1.PackageManifest{}
+	manifestWithSimpleValue := v1alpha1.PackageManifest{
+		ValueDefinitions: map[string]v1alpha1.ValueDefinition{
+			"test": {Type: v1alpha1.ValueTypeText}},
+	}
+	manifestWithRequiredValue := v1alpha1.PackageManifest{
+		ValueDefinitions: map[string]v1alpha1.ValueDefinition{
+			"test": {Type: v1alpha1.ValueTypeText, Constraints: v1alpha1.ValueDefinitionConstraints{Required: true}}},
+	}
+	manifestWithConstraints := v1alpha1.PackageManifest{
+		ValueDefinitions: map[string]v1alpha1.ValueDefinition{
+			"minmaxstr": {
+				Type: v1alpha1.ValueTypeText,
+				Constraints: v1alpha1.ValueDefinitionConstraints{
+					MinLength: 5,
+					MaxLength: 10,
+				},
+			},
+			"minmax": {
+				Type: v1alpha1.ValueTypeNumber,
+				Constraints: v1alpha1.ValueDefinitionConstraints{
+					Min: 5,
+					Max: 10,
+				},
+			},
+			"pattern": {
+				Type: v1alpha1.ValueTypeText,
+				Constraints: v1alpha1.ValueDefinitionConstraints{
+					Pattern: "a{2,3}b+",
+				},
+			},
+			"options": {
+				Type:    v1alpha1.ValueTypeOptions,
+				Options: []string{"foo", "bar"},
+			},
+			"bool": {Type: v1alpha1.ValueTypeBoolean},
+		},
+	}
+	DescribeTable("Validating values",
+		func(manifest v1alpha1.PackageManifest, values map[string]string, valid bool) {
+			err := ValidateResolvedValues(manifest, values)
+			if valid {
+				Expect(err).NotTo(HaveOccurred())
+			} else {
+				Expect(err).To(HaveOccurred())
+			}
+		},
+		Entry("When def is empty and config is empty", emptyManifest, map[string]string{}, true),
+		Entry("When value without def", emptyManifest, map[string]string{"test": "test"}, false),
+		Entry("When value with matching def", manifestWithSimpleValue, map[string]string{"test": "test"}, true),
+		Entry("When required value missing", manifestWithRequiredValue, map[string]string{}, false),
+		Entry("When required value present", manifestWithRequiredValue, map[string]string{"test": "test"}, true),
```

Comment:
Good point!

In my imagination, the `Required` constraint would only check whether a value for that key is present. The reasoning here was that there might be some cases where an empty string might be an allowed value but the patch must always be applied anyways. Indeed, if a package is configured via the UI, a value would be created for each value so the constraint would not do much in this case. 

I understand that this might be confusing from a UI perspective so I am open for discussion in this regard. I will definitely add this as a test case regardless of what _should_ happen.